From 42532f8ba7f9265d74cac2ec2a7709a2b7eb8a0e Mon Sep 17 00:00:00 2001
From: Sarika Sharma <quic_sarishar@quicinc.com>
Date: Mon, 12 Aug 2024 12:50:06 +0530
Subject: [PATCH] wifi: ath12k: Set disable monitor filter to true during rmmod

Currently during rmmod or radio stop, monitor filter set disable
is not set to true.
As monitor filter disable is false, hardware may still write on
buffer that leads to memory corruption.
This memory corruption leading to crash.

However, set tlv_filter.rxmon_disable to true in
ath12k_mac_config_mon_status_default() during radio stop will not lead to
memory corruption.

Patch-dependency: none

Signed-off-by: Sarika Sharma <quic_sarishar@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index 66b2a4f..858077a 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -10983,6 +10983,8 @@ static int ath12k_mac_config_mon_status_default(struct ath12k *ar, bool enable)
 						 HTT_RX_FILTER_TLV_FLAGS_PPDU_START_USER_INFO);
 #endif /* CPTCFG_ATH12K_PPE_DS_SUPPORT */
 
+	} else {
+		tlv_filter.rxmon_disable = true;
 	}
 
 	tlv_filter.offset_valid = false;
-- 
2.34.1

